xenserver/tests/test_tasks.py

```diff
@@ -3,186 +3,265 @@
 """
 
 import pytest
+from testtools.assertions import assert_that
```

Comment:
Side question: Do you have any feelings w.r.t. using the testtools base TestCase class with `self.assertThat` vs this `assert_that` function?

I've been leaning towards the latter lately as I see testtools as the library for matchers in the tests while pytest is the "test runner".